# -*- coding: utf-8 -*-
"""
Created on Mon Sep 12 15:56:02 2022

@author: arman hossain

!python excel.py
"""

import sys

from PyQt5.QtWidgets import (
    QApplication, QDialog, QTableWidget,QWidget, QMainWindow, QMessageBox, QGroupBox, QLabel
)
from PyQt5.uic import loadUi


class Excel(QWidget):
    def __init__(self, parent=None):
        super().__init__(parent)
        loadUi("excel.ui", self)
        self.tableWidget.goto_page(1)
        self.setbuttontext(1)
        self.setWindowTitle("List View")
        self.connect()
        
    def connect(self):
        self.previous_btn.clicked.connect(self.previous_pressed)
        self.next_btn.clicked.connect(self.next_pressed)
        self.plus_btn.clicked.connect(self.current_plus_pressed)
        self.minus_btn.clicked.connect(self.current_minus_pressed)
        # self.current_btn.clicked.connect(self.current_btn_pressed)
        
    def setbuttontext(self,number):
        if number < 1:
            return
        if number == 1:
            self.previous_btn.setEnabled(False)
        else:
            self.previous_btn.setEnabled(True)
        next = number+self.tableWidget.item_per_page
        pre = number-self.tableWidget.item_per_page
        self.current_btn.setText(str(number))
        self.plus_btn.setText(str(next))
        if pre>0:
            self.minus_btn.setEnabled(True)
            self.minus_btn.setText(str(pre))
        else:
            self.minus_btn.setText(str(0))
            self.minus_btn.setEnabled(False)
        
    def current_plus_pressed(self):
        self.tableWidget.goto_page(int(self.plus_btn.text()))
        self.setbuttontext(int(self.plus_btn.text()))
    def current_minus_pressed(self):
        self.tableWidget.goto_page(int(self.minus_btn.text()))
        self.setbuttontext(int(self.minus_btn.text()))
    def next_pressed(self):
        self.tableWidget.goto_page_next()
        self.setbuttontext(int(self.current_btn.text())+1)
    def previous_pressed(self):
        self.tableWidget.goto_page_pre()
        self.setbuttontext(int(self.current_btn.text())-1)
        


if __name__ == "__main__":
    app = QApplication(sys.argv)
    win = Excel()
    win.show()
    sys.exit(app.exec())

